### PR TITLE
APPSRE-6370: Update Terraform AWS provider to version 3.76.0

### PR DESCRIPTION
**Pull request by kwilczynski**
HashiCorp has released a maintenance version of the Terraform AWS provider from the **3.x** family, see

- [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4460-december--8-2022)
- Milestone: [v3.76.0](https://github.com/hashicorp/terraform-provider-aws/milestone/253?closed=1)

This particular release includes two backported changes that are of particular interest to us, as per:

- [release/3.x: Migrate from Keybase's Go OpenPGP crypto fork to ProtonMail's](https://github.com/hashicorp/terraform-provider-aws/pull/27275)
- [release/3.x: Backport #25768; Remove Computed from apply_immediately](https://github.com/hashicorp/terraform-provider-aws/pull/28298)

Thus, update the AWS provider to version **3.76.0** as per the upstream release. Note that this version **WILL NOT** introduce many unexpected changes, as an upgrade to the **4.x** family would most certainly, as this is purely a maintenance release that carries no substantial changes.

Related: [APPSRE-6370](https://issues.redhat.com/browse/APPSRE-6370)

Signed-off-by: Krzysztof Wilczyński [kwilczynski@redhat.com](mailto:kwilczynski@redhat.com)

### CODE REVIEW

**Comment by kwilczynski** [comment]
**To the reviewer:**

This has been tested locally to ensure that there aren't any changes being introduced.

File: qontract-reconcile-base/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN dnf -y install unzip glibc-langpack-en
 ENV LC_ALL=en_US.utf8
 ENV LANG=en_US.utf8
 
-ENV TF_PROVIDER_AWS_VERSIONS="3.75.2"
+ENV TF_PROVIDER_AWS_VERSIONS="3.76.0"
```
Comment (rporres) [inline qontract-reconcile-base/Dockerfile:8]:
```suggestion
ENV TF_PROVIDER_AWS_VERSIONS="3.75.2 3.76.0"
```

Comment (kwilczynski) [reply qontract-reconcile-base/Dockerfile]: @rporres, good idea!

Comment (kwilczynski) [reply qontract-reconcile-base/Dockerfile]: Done.

**Review by rporres — APPROVED** [review]

**Pull request merged**